drivers/gpio/gpio_max32.h

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2023-2024 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_GPIO_GPIO_MAX32_H_
+#define ZEPHYR_DRIVERS_GPIO_GPIO_MAX32_H_
+
+#include <zephyr/device.h>
+
+int gpio_max32_config_pinmux(const struct device *dev, int pin, int afx, int pincfg);
```

Comment:
what's going on here?

Reply:
please, make pinctrl driver self-contained, IIRC only ST has something similar for historical reasons.

Reply:
Thanks, it is done. please see [pinctrl_max32.c](https://github.com/zephyrproject-rtos/zephyr/blob/23a978e6b17827e4655b4a35d006346c00abe57c/drivers/pinctrl/pinctrl_max32.c)